include/lsst/afw/cameraGeom/TransformMap.h

```diff
@@ -199,6 +205,118 @@ class TransformMap final {
     CameraSysFrameIdMap const _frameIds;
 };
 
+
+/**
+ * Helper class used to incrementally construct TransformMap instances.
+ */
+class TransformMap::Builder {
+public:
+
+    /**
+     * Construct an empty builder with no transforms and only the given
+     * coordinate system.
+     */
+    explicit Builder(CameraSys const & reference);
+
+    ///@{
+    /// Builders are copyable, movable, and assignable.
+    Builder(Builder const &);
+    Builder(Builder &&);  // std::vector move construct is not noexcept until C++17
+    Builder & operator=(Builder const &);
+    Builder & operator=(Builder &&);  // std::vector move assignment is not noexcept until C++17
+    ///@}
+
+    ~Builder() noexcept;
+
+    /**
+     * Add a new coordinate system to the builder.
+     *
+     * @param  fromSys   Coordinate system for the arguments to
+     *                   `transform->applyForward`.
+     * @param  toSys     Coordinate system for the return values of
+     *                   `transform->applyForward`.
+     * @param  transform Mapping from `fromSys` to `toSys`.
+     *
+     * @returns `*this` to enable chained calls.
+     *
+     * @throws pex::exceptions::InvalidParameterError  Thrown if the transform
+     *     does not have forward or inverse mapping, or if `fromSys` and
+     *     `toSys` are the same.
+     *
+     * @exceptsafe strong
+     */
+    Builder & connect(CameraSys const & fromSys, CameraSys const & toSys,
+                      std::shared_ptr<geom::TransformPoint2ToPoint2 const> transform);
```

Comment:
I've moved in the implementation of this function instead.  Our standards actually say to *always* pass `shared_ptr` by value, which I think is defensible only from the perspective that it simplifies things for readers who aren't thinking about the cost of an atomic reference count.

But since this is a case where the function is definitely taking a reference to the object I think pass-by-value is appropriate more generally, as per http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#r34-take-a-shared_ptrwidget-parameter-to-express-that-a-function-is-part-owner.